chainer/functions/connection/depthwise_convolution_2d.py

```diff
@@ -0,0 +1,183 @@
+            .reshape((C, B * IY * IX, KY * KX))
+        # (C, D, B*IY*IX), (C, B*IY*IX, KY*KX) -> (C, D, KY*KX)
+        gW_ = xp.matmul(gy_, c_)
+        gW = gW_.reshape((C, D, KY, KX)).transpose(1, 0, 2, 3)
+        gW = gW.astype(W.dtype, copy=False)
+
+        w_ = W.transpose(1, 2, 3, 0).reshape((C, KY * KX, D))
+        # (C, KY*KX, D), (C, D, B*IY*IX) -> (C, KY*KX, B*IY*IX)
+        gcol = xp.matmul(w_, gy_).reshape((C, KY, KX, B, IY, IX))
+        gcol = gcol.astype(x.dtype, copy=False)
+        gcol = xp.rollaxis(gcol, 3)
+
+        if xp is numpy:
+            gx = conv.col2im_cpu(gcol, self.sy, self.sx,
+                                 self.ph, self.pw, h, w)
+        else:
+            gx = conv.col2im_gpu(gcol, self.sy, self.sx,
+                                 self.ph, self.pw, h, w)
+
+        if b is None:
+            return gx, gW
+        else:
+            gy = xp.rollaxis(gy, 1, 4)
+            gb = gy.sum(axis=(0, 1, 2))
+            return gx, gW, gb
+
+
+def depthwise_convolution_2d(x, W, b=None, stride=1, pad=0):
+    """Two-dimensional depthwise convolution function.
+
+    This is an implementation of two-dimensional depthwise convolution.
+    It takes three variables: the input image ``x``, the filter weight ``W``,
+    and the bias vector ``b``.
+
+    Notation: here is a notation for dimensionalities.
+
+    - :math:`n` is the batch size.
+    - :math:`c_I` is the number of the input.
+    - :math:`c_M` is the number of the channel multiplier.
+    - :math:`h` and :math:`w` are the height and width of the input image,
+      respectively.
+    - :math:`h_O` and :math:`w_O` are the height and width of the output image,
+      respectively.
+    - :math:`k_H` and :math:`k_W` are the height and width of the filters,
+      respectively.
+
+    Args:
+        x (chainer.Variable or :class:`numpy.ndarray` or cupy.ndarray):
+            Input variable of shape :math:`(n, c_I, h, w)`.
+        W (~chainer.Variable): Weight variable of shape
+            :math:`(c_M, c_I, k_H, k_W)`.
+        b (~chainer.Variable):
+            Bias variable of length :math:`c_M * c_I` (optional).
+        stride (int or pair of ints): Stride of filter applications.
+            ``stride=s`` and ``stride=(s, s)`` are equivalent.
+        pad (int or pair of ints): Spatial padding width for input arrays.
+            ``pad=p`` and ``pad=(p, p)`` are equivalent.
+
+
+    Returns:
+        ~chainer.Variable:
+            Output variable. The shape is :math:`(n, c_I * c_M, h_O, w_O)`.
+
+    Like ``Convolution2D``, ``DepthwiseConvolution2D`` function computes
+    correlations between filters and patches of size :math:`(k_H, k_W)` in
+    ``x``.
+    But unlike ``Convolution2D``, ``DepthwiseConvolution2D`` not adds up
+    between input channels of filters but concatenates.
+    For that reason, the shape of outputs of depthwise convolution are
+    :math:`(n, c_I * c_M, h_O, w_O)`, :math:`c_M` called channel_multiplier.
```

Comment:
For that reason, the shape of outputs of depthwise convolution are :math:`(n, c_I * c_M, h_O, w_O)`. :math:`c_M`  is called *channel multiplier*.